### DISCUSSION TITLE
Setup a proper Code Coverage system in new codebase

### DISCUSSION THREAD

**Discussion by heliocastro** [eclipse-sw360/sw360#2452] [General]
SW360 has loosing build time and not even generating proper results with the current maven plugin.
The results are not stored or used absolutely anywhere.

A better plan should be defined with the new codebase.

**Comment by iihimanshuu** [comment]
We can improve build efficiency by adopting JaCoCo for reliable code coverage, integrating it into pom.xml, and automating reporting with GitHub Actions. Results can be stored and visualized using platforms like Codecov or SonarQube, ensuring actionable insights and better quality tracking.
